### ISSUE TITLE
Error: --mode is missing from device

### ISSUE THREAD

**Issue by Schnabulator** [sbs20/scanservjs#307]
Hi, 
using CLI everything works  fine with my Umax Astra 1600U. 
But the Webinterface scanservjs can't find the device.
Output from systemctl -e -u scanservjs.service:

`Jun 15 13:33:33 raspberrypi server.js[3389]: [2021-06-15T11:33:33.663Z] DEBUG (Devices): Device list:  device `umax1220u:libusb:001:004' is a UMAX Astra 2000U flatbed sca
Jun 15 13:33:33 raspberrypi server.js[3389]: [2021-06-15T11:33:33.750Z] DEBUG (Devices): Device features:
Jun 15 13:33:33 raspberrypi server.js[3389]: All options specific to device `umax1220u:libusb:001:004':
Jun 15 13:33:33 raspberrypi server.js[3389]:     --resolution auto||75|150|300|600dpi [75]
Jun 15 13:33:33 raspberrypi server.js[3389]:         Sets the resolution of the scanned image.
Jun 15 13:33:33 raspberrypi server.js[3389]:     --gray[=(yes|no)] [no]
Jun 15 13:33:33 raspberrypi server.js[3389]:         Do a grayscale rather than color scan
Jun 15 13:33:33 raspberrypi server.js[3389]:     -l 0..228.6mm [0]
Jun 15 13:33:33 raspberrypi server.js[3389]:         Top-left x position of scan area.
Jun 15 13:33:33 raspberrypi server.js[3389]:     -t 0..298.027mm [0]
Jun 15 13:33:33 raspberrypi server.js[3389]:         Top-left y position of scan area.
Jun 15 13:33:33 raspberrypi server.js[3389]:     -x 0..228.6mm [228.6]
Jun 15 13:33:33 raspberrypi server.js[3389]:         Width of scan-area.
Jun 15 13:33:33 raspberrypi server.js[3389]:     -y 0..298.027mm [298.027]
Jun 15 13:33:33 raspberrypi server.js[3389]:         Height of scan-area.
Jun 15 13:33:33 raspberrypi server.js[3389]: [2021-06-15T11:33:33.755Z] ERROR (Devices): Ignoring umax1220u:libusb:001:004. Error: --mode is missing from device`

Where do i have to set the mode? Hope somebody can help me. :)

**Comment by sbs20** [comment]
Oh! I have never seen that before. I will implement a fix for that but it may take a short while.

Background - scanservjs (well, me) is making the assumption that `--mode` is always present - except it's not for your scanner. Thanks for reporting.

**Comment by Schnabulator** [comment]
Nice, thanks a lot. Is there a workaround like configuring the scanner manually?

Thank you in advance!

**Comment by sbs20** [comment]
Not an easy or sensible one, no, sorry

**Comment by sbs20** [comment]
Out of interest, did you change

```
All options specific to device 'umax1220u:libusb:001:004'
```

And replace the backtick with an apostrophe? (presumably in order to format the output in GitHub?)

**Comment by Schnabulator** [comment]
Yes i changed that. I never posted "code" in github before, so i tried a lot to achieve correct formatting
And I thought using an apostrophe instead of the backtick is fine

**Comment by sbs20** [comment]
Don't worry all good! I was just creating a unit test to run against the output but it was failing because the backtick was missing.

**Comment by sbs20** [comment]
I believe this is now fixed in v2.16.0 - if you just run the one line install again then it will update the install and should hopefully just work.

**Comment by Schnabulator** [comment]
Nice work! Thanks a lot. Now it works perfectly. :)
